src/mono/wasm/runtime/assets.ts

```diff
@@ -38,34 +52,70 @@ export async function mono_download_assets(): Promise<void> {
     if (runtimeHelpers.diagnosticTracing) console.debug("MONO_WASM: mono_download_assets");
     runtimeHelpers.maxParallelDownloads = runtimeHelpers.config.maxParallelDownloads || runtimeHelpers.maxParallelDownloads;
     try {
-        const download_promises: Promise<AssetEntry | undefined>[] = [];
+        const buffer_promises: Promise<AssetEntry | undefined>[] = [];
         // start fetching and instantiating all assets in parallel
-        for (const asset of runtimeHelpers.config.assets!) {
-            if (!asset.pending && !skipDownloadsAssetTypes[asset.behavior]) {
-                download_promises.push(start_asset_download(asset));
+        for (const a of runtimeHelpers.config.assets!) {
+            const asset: AssetEntryInternal = a;
+            if (!skipInstantiateAssetTypes[asset.behavior]) {
+                expected_instantiated_assets_count++;
+            }
+            if (!skipDownloadsAssetTypes[asset.behavior]) {
+                const dontTouchBuffer = skipBufferAssetTypes[asset.behavior];
+                expected_downloded_assets_count++;
+                if (asset.pending) {
+                    asset.internalPending = asset.pending;
```

Comment:
Do IIUC, that `asset.pending` is not being set anywhere right now, but the intention is that it can be set for some kinda "pre-loaded asset", or some "asset with custom loading" - so we won't attempt to fetch it?

Also, is `assert.InternalPending` being used because `asset.pending` could be modified by someone once `mono_download_assets` has started executing?

Reply:
the `pending` is a contract with Blazor, their cache is giving that out. The shape of it is pre-existing design.
We have  `internalPending` which is almost the same semantics, but not 100%. So we disconnected those concepts with @maraf yesterday.

Reply:
Can we make that difference clearer in code?

Reply:
Maybe by naming the objects accordingly

Reply:
any hint ?

Reply:
we settled on `pendingDownload?: LoadingResource` and doc